### PR TITLE
Update llpcElfWriter

### PR DESCRIPTION
**Pull request by jiaolu**
Add two new functions GetSectiongetSectionTextShader and
updateSymbolsBySectionIndex

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit aef3a667637aa77c72ea5fe0e6bcfc198fc439db
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 20bc469e9806bb60b22559f727c1321354091898</li>
<li>SPVGEN: dev c054813a9a894e32aaaa04c6717a667c15f60cfd</li>
<li>XGL: dev feb87bf013f2b4bd16f7a1da39a5a97fdaa7470d</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev d3f03b36d914d6f4c4d27709bf29b89c4803c9cf</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 22981/43895 (52.4%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 20914/43895 (47.6%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 22980/43895 (52.4%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 20915/43895 (47.6%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 21622/43874 (49.3%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 22252/43874 (50.7%)</li>
</ul></details>


</ul></details>

File: lgc/patch/Gfx9ConfigBuilder.cpp [hunk 1]
```diff
@@ -844,7 +844,8 @@ void ConfigBuilder::buildVsRegConfig(ShaderStage shaderStage, T *pConfig) {
   bool enableXfb = resUsage->inOutUsage.enableXfb;
   if (shaderStage == ShaderStageCopyShader) {
     // NOTE: For copy shader, we use fixed number of user data registers.
-    SET_REG_FIELD(&pConfig->vsRegs, SPI_SHADER_PGM_RSRC2_VS, USER_SGPR, lgc::CopyShaderUserSgprCount);
+    auto copyShaderUserSprgCount = lgc::CopyShaderUserSgprCount;
```
Comment (amdrexu) [inline lgc/patch/Gfx9ConfigBuilder.cpp:847]: Better to make a separate pull request for this change. It is different from others. Also, better to add explanations in the commit message to describe why you do this since it is very strange for external audience.

**Review by amdrexu — APPROVED** [review]

**Comment by jiaolu** [comment]
separate to two commits

**Comment by amdvlk-admin** [comment]
### Test summary for commit f9535efed1f1897d548543248b188296a65a482b
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev c9937b277c491a55c689eec7cdd48fb238b8004c</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev c6c90450fb1abf5f414acf1e38a0f51a72c426c1</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 0510352269ff92dabcb91c46b1ad30b61299a7e7</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 24006/43895 (54.7%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 19889/43895 (45.3%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 24005/43895 (54.7%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 19890/43895 (45.3%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 22647/43874 (51.6%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 21227/43874 (48.4%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2c99191ee80006b43534590f4d4737034d047e05
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev c9937b277c491a55c689eec7cdd48fb238b8004c</li>
<li>SPVGEN: dev e1f1881d84cd1c7a6c1d661cbac0e619b4e110c1</li>
<li>XGL: dev c6c90450fb1abf5f414acf1e38a0f51a72c426c1</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 0510352269ff92dabcb91c46b1ad30b61299a7e7</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/131664)</summary><ul><li>Built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix</li>
<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 24006/43895 (54.7%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 19889/43895 (45.3%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 24005/43895 (54.7%)</li>
<li>Failed: 0/43895 (0.0%)</li>
<li>Skipped: 19890/43895 (45.3%)</li>
</ul></details>

<details open><summary>Rhel 8.2, Gfx8</summary><ul>

<li>Passed: 22647/43874 (51.6%)</li>
<li>Failed: 0/43874 (0.0%)</li>
<li>Skipped: 21227/43874 (48.4%)</li>
</ul></details>


</ul></details>
